core/src/test/java/org/elasticsearch/index/query/ConstantScoreQueryBuilderTests.java

```diff
@@ -67,15 +68,15 @@ public void testFilterElement() throws IOException {
     }
 
     /**
-     * test that multiple "filter" elements causes {@link ParsingException}
+     * test that multiple "filter" elements causes {@link JsonParseException}
      */
     public void testMultipleFilterElements() throws IOException {
         String queryString = "{ \"" + ConstantScoreQueryBuilder.NAME + "\" : {\n" +
                                     "\"filter\" : { \"term\": { \"foo\": \"a\" } },\n" +
                                     "\"filter\" : { \"term\": { \"foo\": \"x\" } },\n" +
                             "} }";
-        ParsingException e = expectThrows(ParsingException.class, () -> parseQuery(queryString));
-        assertThat(e.getMessage(), containsString("accepts only one 'filter' element"));
+        JsonParseException e = expectThrows(JsonParseException.class, () -> parseQuery(queryString));
```

Comment:
Great idea; I love it. I'll revert the tests to their previous state, add the compile time constant and skip the tests as you've suggested.